### PR TITLE
Feat(engine): update to bcrypt for windows. 

### PR DESCRIPTION
**Pull request by tensor-programming**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

Updates the windows random API to bcrypt from wincrypt which is deprecated. 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Ran the test suite.  

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: engine/random/c_src/rng.c [hunk 1]
```diff
@@ -28,13 +27,21 @@ uint8_t os_random_secrandom(uint8_t *buf, size_t len)
 #elif defined(USE_SECRANDOM)
     return SecRandomCopyBytes(kSecRandomDefault, len, buf) == errSecSuccess ? 0 : 1;
 #elif defined(USE_CRYPTGENRANDOM)
-    HCRYPTPROV rng;
-    if (CryptAcquireContext(&rng, NULL, NULL, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT) == 0)
+    BCRYPT_ALG_HANDLE algo_handle;
+    NTSTATUS ret1 = BCryptOpenAlgorithmProvider(&algo_handle, BCRYPT_RNG_ALGORITHM,
```
Comment (rajivshah3) [inline engine/random/c_src/rng.c:31]: Is it necessary to initialize the provider if `algo_handle` is not being passed to `BCryptGenRandom`?

Comment (tensor-programming) [reply engine/random/c_src/rng.c:31]: Did I forget to pass it?

**Comment by tensor-programming** [comment]
@nothingismagick the formatting is all from stronghold not from parti.

File: engine/random/c_src/rng.c (hunk 1 above)
Comment (rajivshah3) [reply engine/random/c_src/rng.c]: > Did I forget to pass it?

Nope, I think it's just not needed here. The [docs](https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom) say the `hAlgorithm` parameter should be `NULL` when using `BCRYPT_USE_SYSTEM_PREFERRED_RNG` so it seems like you don't need a handle at all

Comment (tensor-programming) [reply engine/random/c_src/rng.c]: Figured that one out after I wrote it.  Had it removed in an earlier version but added it back in by accident but forgot to actually put it in as a argument.

**Pull request merged**